lab_5_scrapper/scrapper.py

```diff
@@ -1,82 +1,196 @@
 """
 Crawler implementation
 """
+import datetime
+import json
+import re
+import shutil
+import time
+from pathlib import Path
+from random import randint
 from typing import Pattern, Union
 
+import requests
+from bs4 import BeautifulSoup
+
+from core_utils.article.article import Article
+from core_utils.article.io import to_meta, to_raw
+from core_utils.config_dto import ConfigDTO
+from core_utils.constants import (ASSETS_PATH, CRAWLER_CONFIG_PATH,
+                                  NUM_ARTICLES_UPPER_LIMIT,
+                                  TIMEOUT_LOWER_LIMIT, TIMEOUT_UPPER_LIMIT)
+
+
+class IncorrectSeedURLError(Exception):
+    """
+    Seed URL does not match standard pattern or does not correspond to the target website
+    """
+    pass
+
+
+class NumberOfArticlesOutOfRangeError(Exception):
+    """
+    Total number of articles is out of the needed range
+    """
+    pass
+
+
+class IncorrectNumberOfArticlesError(Exception):
+    """
+    Total number of articles to parse is not integer
+    """
+    pass
+
+
+class IncorrectHeadersError(Exception):
+    """
+    Headers are not in a form of dictionary
+    """
+    pass
+
+
+class IncorrectEncodingError(Exception):
+    """
+    Encoding must be specified as a string
+    """
+    pass
+
+
+class IncorrectTimeoutError(Exception):
+    """
+    Incorrect timeout value
+    """
+    pass
+
+
+class IncorrectVerifyError(Exception):
+    """
+    Incorrect verify certificate value
+    """
+    pass
+
 
 class Config:
     """
     Unpacks and validates configurations
     """
 
+    seed_urls: list[str]
+    total_articles_to_find_and_parse: int
+    headers: dict[str, str]
+    encoding: str
+    timeout: int
+    verify_certificate: bool
+    headless_mode: bool
+
     def __init__(self, path_to_config: Path) -> None:
         """
         Initializes an instance of the Config class
         """
-        pass
+        self.path_to_config = path_to_config
+        self._validate_config_content()
```

Comment:
in init:

1. you need to extract config's content with a function
2. run validation using DTO from the previuos step
3. fill attributes of config and continue work